### ISSUE TITLE
`Delivery.tag` is the wrong type in typings

### ISSUE THREAD

**Issue by ps2goat** [amqp/rhea#82]
[This line](https://github.com/amqp/rhea/blob/master/typings/session.d.ts#L20) is incorrect, according to it's usage [here](https://github.com/amqp/rhea/blob/master/typings/link.d.ts#L68).

It looks like the `Delivery.tag` type should be `Buffer | string`.  I'm currently doing this to get it to a string (which is what my value really is), because TypeScript:

`<string>(result.tag as any)`

**Comment by amarzavery** [comment]
makes sense. @grs - I remember having this conversation with you. Can the delivery tag be a string OR Buffer?

**Comment by ps2goat** [comment]
It works, and I'd prefer to keep using it as a string.  It's how I'm tracking if a message sent was accepted.

**Comment by ps2goat** [comment]
I would be ok with that as long as there is another way to correlate the message id during a sender message result event (`accepted`, `released`, or `rejected`).  Right now, the delivery tag seemed to be the way to do that, so I set the tag equal to the unique message id.

## TLDR; 
If you could pass the sent `Message` back on the `EventContext` for sender message events, that would do what I want (a way to correlate the message that was accepted/rejected/released with the message I am looking for in a message-specific listener) and keep things to spec.  This would actually be preferred because it's more straight-forward.

## What I'm currently doing
I'm abstracting and promisifying the `send` call, so a caller can do `const wasSuccessful: boolean = await myClassInstance.send(msgObj);` and essentially get the result of sending the message.  I want the server to accept it so the calling app can ensure that the message was accepted without having to implement the listeners themselves.  We have several apps that will need this functionality, so it makes sense to abstract it.

So after calling `sender.send` in rhea, passing in my message and the message ID as the `tag` param, I wire up the listener for the 3 events and resolve/reject in those events (similar to @amarzavery's examples of what is in the Azure Service Bus sdks), and remove those listeners.  I then compare the expected messageId (the one I generated and assigned to the `Message`) to the Delivery's `tag` property.

**Comment by grs** [comment]
You could still correlate even if you had to send it as Buffer, it just means you need to call toString() on it. Passing the Message with subsequent events means having to hold on to the message itself which I would rather not do.

However I have no objection to the typescript typing being changed for the tag on Delivery, since it can at present be a string or Buffer. For Deliveries received from peer it will always be a Buffer.

**Comment by ps2goat** [comment]
That would be fine, then.  

Right now, it's returned as a string in the Delivery object returned from `send`, so that would potentially be broken functionality for others. It would be good as a milestone for the next major version.
